### PR TITLE
manifest: add compatibility test for Permissions and Trust serialization

### PR DESCRIPTION
**Pull request by AnnaShaleva**
We already have tests for Permission deserialisation, so port the first part of https://github.com/neo-project/neo/pull/2948.

### CODE REVIEW

File: pkg/smartcontract/manifest/manifest_test.go [hunk 1]
```diff
@@ -305,6 +305,25 @@ func TestManifestToStackItem(t *testing.T) {
...
+		c := NewManifest("Test")
+		require.NoError(t, json.Unmarshal([]byte(mJSON), c))
+
+		si, err := c.ToStackItem()
+		require.NoError(t, err)
+		actual := new(Manifest)
+		require.NoError(t, actual.FromStackItem(si))
+		require.False(t, actual.Permissions[0].Contract.Type == PermissionWildcard)
```
Comment (roman-khimov) [inline pkg/smartcontract/manifest/manifest_test.go:319]: Why not `require.NotEqual` here?

File: pkg/smartcontract/manifest/manifest_test.go [hunk 2]
```diff
@@ -305,6 +305,25 @@ func TestManifestToStackItem(t *testing.T) {
...
+		require.True(t, actual.Permissions[0].Methods.IsWildcard())
+		require.True(t, actual.Permissions[1].Contract.Type == PermissionWildcard)
```
Comment (roman-khimov) [inline pkg/smartcontract/manifest/manifest_test.go:321]: And `require.Equal` here?

File: pkg/smartcontract/manifest/manifest_test.go (hunk 1 above)
Comment (AnnaShaleva) [reply pkg/smartcontract/manifest/manifest_test.go]: Fixed.

File: pkg/smartcontract/manifest/manifest_test.go (hunk 2 above)
Comment (AnnaShaleva) [reply pkg/smartcontract/manifest/manifest_test.go]: Fixed. Just copy-pasted.

**Review by roman-khimov — APPROVED** [review]

**Pull request merged**